File: solutions/485. Max Consecutive Ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        #recent attempt, time O(N), space O(1)
        # i = -1
        # ans = 0
        # for j, num in enumerate(nums):
        #     if num == 0:
        #         i = j
        #     else:
        #         ans = max(ans, j-i)
        # return ans
    
        #old attempt, time O(N), space O(1)
        res = 0
        count = 0        
        for n in nums:
            if n:
                count += 1
            else:
                res = max(res, count)
                count = 0
        return max(res, count)

File: solutions/485. Max Consecutive Ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        // better, time O(n), space O(1)
        int ans = 0, count = 0;
        for (auto num : nums) {
            if (num) {
                count++;
                ans = max(ans, count);
            }
            else count = 0;
        }
        return ans;
​
        // // 1st attempt, time O(n), space O(1)
        // int j = -1;
        // int ans = 0;
        // int len = nums.size();
        // for (int i = 0; i < len; ++i) {
        //     if (nums[i] == 0) {
        //         ans = max(ans, i-j-1);
        //         j = i;
        //     }
        // }
        // ans = max(ans, len-j-1);
        // return ans;
    }
};
